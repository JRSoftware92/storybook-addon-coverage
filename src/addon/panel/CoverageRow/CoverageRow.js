import React from 'react';

const getClassNameByPercent = (percent, passThreshold, failThreshold) => {
  if (percent >= passThreshold) {
    return 'coverage-complete';
  } else if (percent <= failThreshold) {
    return 'coverage-none';
  }

  return 'coverage-partial';
};

const CoverageRow = ({ label, passThreshold, failThreshold, rowData: { total, covered, skipped, pct } }) => (
  <tr id={`coverage-row-${label}`} className={getClassNameByPercent(pct, passThreshold, failThreshold)}>
    <td id="cell-label" className="bold">{label}</td>
    <td id="cell-covered">{covered}</td>
    <td id="cell-skipped" className={skipped ? 'bold' : ''}>{skipped}</td>
    <td id="cell-total">{total}</td>
    <td id="cell-pct">{pct}%</td>
  </tr>
);

export default CoverageRow;
