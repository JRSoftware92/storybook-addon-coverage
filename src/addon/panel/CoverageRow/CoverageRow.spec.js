import React from 'react';
import { shallow } from 'enzyme';

import CoverageRow from './CoverageRow';

const getWrapper = props => shallow(<CoverageRow {...props} />);

const rowData = {
  total: 3,
  covered: 3,
  skipped: 0,
  pct: 70,
};

const props = {
  passThreshold: 80,
  failThreshold: 50,
  label: 'Row Label',
  rowData,
};

describe('CoverageRow', () => {
  test('renders correctly for partial coverage', () => {
    const wrapper = getWrapper(props);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly for passing coverage', () => {
    const wrapper = getWrapper({
      ...props,
      rowData: {
        ...rowData,
        pct: 100,
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly for failing coverage', () => {
    const wrapper = getWrapper({
      ...props,
      rowData: {
        ...rowData,
        pct: 0,
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
