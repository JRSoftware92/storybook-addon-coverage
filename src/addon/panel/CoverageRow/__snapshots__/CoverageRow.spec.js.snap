// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coverage Table Row bolds skipped entries 1`] = `"<tr id=\\"coverage-row-Row Label\\" class=\\"coverage-partial\\"><td id=\\"cell-label\\" class=\\"bold\\">Row Label</td><td id=\\"cell-covered\\">3</td><td id=\\"cell-skipped\\" class=\\"bold\\">1</td><td id=\\"cell-total\\">3</td><td id=\\"cell-pct\\">70%</td></tr>"`;

exports[`Coverage Table Row displays a green background for passing coverage 1`] = `"<tr id=\\"coverage-row-Row Label\\" class=\\"coverage-complete\\"><td id=\\"cell-label\\" class=\\"bold\\">Row Label</td><td id=\\"cell-covered\\">3</td><td id=\\"cell-skipped\\" class=\\"\\">0</td><td id=\\"cell-total\\">3</td><td id=\\"cell-pct\\">100%</td></tr>"`;

exports[`Coverage Table Row displays a red background for failing coverage 1`] = `"<tr id=\\"coverage-row-Row Label\\" class=\\"coverage-none\\"><td id=\\"cell-label\\" class=\\"bold\\">Row Label</td><td id=\\"cell-covered\\">3</td><td id=\\"cell-skipped\\" class=\\"\\">0</td><td id=\\"cell-total\\">3</td><td id=\\"cell-pct\\">0%</td></tr>"`;

exports[`Coverage Table Row displays a yellow background for partial coverage 1`] = `"<tr id=\\"coverage-row-Row Label\\" class=\\"coverage-partial\\"><td id=\\"cell-label\\" class=\\"bold\\">Row Label</td><td id=\\"cell-covered\\">3</td><td id=\\"cell-skipped\\" class=\\"\\">0</td><td id=\\"cell-total\\">3</td><td id=\\"cell-pct\\">70%</td></tr>"`;
